apps/src/netsim/NetSimHeartbeat.js

```diff
@@ -133,12 +141,30 @@ NetSimHeartbeat.prototype.setBeatInterval = function (intervalMs) {
 };
 
 /**
+ * Set a handler to call if this heartbeat is unable to update its remote
+ * storage representation.  Can be used to go into a recovery mode,
+ * acknowledge disconnect, and/or attempt an auto-reconnect.
+ * @param {function} onFailedHeartbeat
+ */
+NetSimHeartbeat.prototype.setFailureCallback = function (onFailedHeartbeat) {
+  this.onFailedHeartbeat_ = onFailedHeartbeat;
+};
+
+/**
  * Updates own row on regular interval, as long as something's making
  * it tick.
  */
 NetSimHeartbeat.prototype.tick = function () {
   if (Date.now() - this.time_ > this.intervalMs_) {
     this.time_ = Date.now();
-    this.update();
+    this.update(function (success) {
+      if (!success) {
```

Comment:
Sure.  I wasn't familiar with that convention; I was sort of following what Geoffrey set up with the storage table wrapper methods so callbacks could just be passed all the way down, but I'd be willing to switch things over.

Do you have a reference for this convention?  Is there a certain type of object that gets passed to `err` (like an actual `Error`)?  If there's no error, do these callbacks expect `null` or `undefined`?